fendermint/vm/interpreter/src/fvm/state/exec.rs

```diff
@@ -350,3 +351,39 @@ fn check_error(e: anyhow::Error) -> (ApplyRet, ActorAddressMap) {
     };
     (ret, Default::default())
 }
+
+pub struct ElapsedExecution<'a, DB>
+where
+    DB: Blockstore + Clone + 'static,
+{
+    state: &'a mut FvmExecState<DB>,
+}
+
+impl<'a, DB> ElapsedExecution<'a, DB>
+where
+    DB: Blockstore + Clone + 'static,
+{
+    pub fn new(state: &'a mut FvmExecState<DB>) -> Self {
+        Self { state }
+    }
+
+    pub fn execute_implicit(
+        &mut self,
+        msg: Message,
+    ) -> anyhow::Result<(ApplyRet, ActorAddressMap, Duration)> {
+        let start = Instant::now();
+        let result = self.state.execute_implicit(msg)?;
+        let duration = start.elapsed();
+        Ok((result.0, result.1, duration))
+    }
+
+    pub fn execute_explicit(
+        &mut self,
+        msg: Message,
+    ) -> anyhow::Result<(ApplyRet, ActorAddressMap, Duration)> {
+        let start = Instant::now();
+        let result = self.state.execute_explicit(msg)?;
+        let duration = start.elapsed();
+        Ok((result.0, result.1, duration))
+    }
+}
```

Comment:
Introducing a new type seems a bit heavy-handed to me. It's only used in 2 sites, and I think it's acceptable to add a helper function to the observability crate -- this would allow us to time any closure.

```rust
fn measure_time<F, T>(f: F) -> (T, std::time::Duration)
where
    F: FnOnce() -> T,
{
    let start = Instant::now();
    let result = f();
    let duration = start.elapsed();
    (result, duration)
}
```

Reply:
Yeah I agree. This change was a reaction to this [comment here](https://github.com/consensus-shipyard/ipc/pull/1064#discussion_r1679114289)